drivers/input/keyboard/brain-kbd-i2c.c

```diff
@@ -97,28 +98,62 @@ static bool detect_key(struct bk_i2c_data *kbd, u8 keycode)
 	}
 
 	if (kbd->symbol) {
+		if ((0 == BK_IS_PRESSED(keycode)) &&
+				(false == kbd->symbol_flag[keycode & 0x3f])) {
```

Comment:
[1] The expr we're mainly focusing on should be on the left side of the binary operator.
```suggestion
		if ((BK_IS_PRESSED(keycode) == 0) &&
				(kbd->symbol_flag[keycode & 0x3f] == false)) {
```